MzML/Mzml.cs

```diff
@@ -363,11 +365,11 @@ private static int GetOneBasedPrecursorScanNumber(Generated.mzMLType _mzMLConnec
             do
             {
                 oneBasedSpectrumNumber--;
-            } while (!precursorID.Equals(_mzMLConnection.run.spectrumList.spectrum[oneBasedSpectrumNumber - 1].id));
+            } while (!precursorID.Equals(_mzMLConnection.run.spectrumList.spectrum[oneBasedSpectrumNumber].id));
```

Comment:
What if there are two ms1 scans, where the first one is the actual precursor? Could you instead check if the scan  id matches?